### PR TITLE
Linked cameras in subplots

### LINKED ISSUE #253: Linked cameras in subplots
We should implement a way to link cameras across subplots for comparing meshes. for example:

```py
import pyvista as pv
from pyvista import examples
import pyacvd

# download mesh
mesh = examples.download_cow()

clus = pyacvd.Clustering(mesh)
# mesh is not dense enough for uniform remeshing
clus.subdivide(3)
clus.cluster(20000)

# remesh
remesh = clus.create_mesh()

p = pv.Plotter(notebook=0, shape=(1,2))
p.add_mesh(mesh, show_edges=True, color=True)
p.subplot(0,1)
p.add_mesh(remesh, color=True, show_edges=True)
p.show()
```

ezgif com-video-to-gif

Then it might be useful to add a function in Tetgen, PyMeshFix, and PyACVD called `compare` that would plot a side-by-side of the original mesh and the repaired/fixed/remeshed mesh with the cameras linked.

### PR DESCRIPTION
**Pull request by GuillaumeFavelier**
This PR fixes #253 by introducing two new functions called `link_views()` and `unlink_views()`.

Following #253 example, this a code snippet using those functions:

```py
import pyvista as pv
from pyvista import examples
import pyacvd

# download mesh
mesh = examples.download_cow()

clus = pyacvd.Clustering(mesh)
# mesh is not dense enough for uniform remeshing
clus.subdivide(3)
clus.cluster(20000)

# remesh
remesh = clus.create_mesh()

p = pv.Plotter(notebook=0, shape=(2,2))
p.add_mesh(mesh, show_edges=True, color=True)
p.subplot(0,1)
p.add_mesh(remesh, color=True, show_edges=True)
p.subplot(1,0)
p.add_mesh(mesh, color=True, show_edges=False)
p.subplot(1,1)
p.add_mesh(remesh, color=True, show_edges=False)
p.link_views() # link all the views
p.unlink_views([2, 3]) # unlink the last 2 views by giving indices
p.show()
```

output

As a side note, I noticed that with this implementation enabling Eye-Dome lighting gives a white window.

### CODE REVIEW

**Comment by banesullivan** [comment]
This is awesome! Thanks for talking this @GuillaumeFavelier!

**Comment by akaszynski** [comment]
Agreed, this is really useful.  Thanks!

**Comment by banesullivan** [comment]
> As a side note, I noticed that with this implementation enabling Eye-Dome lighting gives a white window.

Hm, that's strange, I'll tcheck out this branch and play with the EDL rendering sometime today

**Comment by banesullivan** [comment]
I'm actually seeing the whiteout view issue on the master branch and it's quite weird... all of the current subplot examples work fine but this cow example yield the whiteout views. I'm not sure what could be happening here. That bug seems to be unrelated to these changes, so I'll make a new issue

**Comment by akaszynski** [comment]
Great work!

**Comment by GuillaumeFavelier** [comment]
I changed to `decimate_boundary()` instead but if you have any ideas of a better comparison, don't hesitate to tell me.

File: examples/02-plot/plot-linked.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+"""
+Linked Views in Subplots
+~~~~~~~~~~~~~~~~
```
Comment (banesullivan) [inline examples/02-plot/plot-linked.py]: Would you please make sure there are as many `~` as characters in the title line

**Comment by banesullivan** [comment]
@GuillaumeFavelier: This looks great! The use of `decimate_boundary()` is perfect for this to have a dataset with drastic differences. Perhaps these other examples that have comparison would benefit from this:

- [Decimation](https://docs.pyvista.org/examples/01-filter/decimate.html)
- [Using Common Filters](https://docs.pyvista.org/examples/01-filter/using-filters.html#sphx-glr-examples-01-filter-using-filters-py)

**Comment by banesullivan** [comment]
@GuillaumeFavelier - looks good to me. Let me know when you're finished and we can merge this

**Comment by GuillaumeFavelier** [comment]
I'm pretty much done, I was just waiting for the CIs basically.

**Comment by banesullivan** [comment]
Cool, I'll come back to this in a bit when the CIs give green checks and merge

**Pull request merged**